### PR TITLE
Impl ci

### PR DESCRIPTION
**Pull request by mathiassmichno**

### CODE REVIEW

File: cha/implementaion/initial/ci.tex [hunk 1]
```diff
@@ -1 +1,36 @@
...
+\end{enumberate}
+
+To carry out these tasks we use a piece of software called Jenkins\footnote{\url{https://jenkins.io/}}, which is a build automation server.
+The build and build process, our Android app in this case, is then defined in a \textit{Jenkinsfile}, which is a recipe with steps to follow for the Jenkins server.
+In our case, the output of the continuous integration using a Jenkins server, are test and linting reports, documentation, and the aforementioned packaged apk--file.
+
+The process of developing on our Android app, is based upon a workflow revolving around source code management using \textit{git}\footnote{\url{https://git-scm.com/}}, and project management using Phabricator\footnote{\url{https://www.phacility.com/}} and the Github\footnote{\url{https://github.com}} feature \textit{pull requests}.
+This development process will be further explained and discussed in \cnameref{cha:development_method}.
+
+We set up the continuous integration, such that any pull request and update to the master branch is activated the automated build system on a in--house Jenkins server.
```
Comment (ThomasPilgaard) [inline cha/implementaion/initial/ci.tex]: Er det ikke "is activated" -> "activates", eller misforstår jeg meningen med sætningen?

**Review by ThomasPilgaard — DISMISSED** [review]
Stefan vil højest sandsynlig have en figur som giver et overblik over CI flowet.

File: cha/implementaion/initial/ci.tex [hunk 2]
```diff
@@ -1 +1,52 @@
-\section{Continous Integration}
+\section{Automated System Building with Continuous Integration}
+%We want to be assisted in integrating newly developed features etc.
+%Also, automate tests (dev env / target) and docs
+%Android APK availability
+%Hand in hand with agile
+While implementing our Android application, it is important that we do not get disconnected from the end product, albeit an unfinished one.
```
Comment (TriXXeD) [inline cha/implementaion/initial/ci.tex]: Do you mean that it is important that we do not lose focus? not sure what you mean by disconnected

File: cha/implementaion/initial/ci.tex [hunk 3]
```diff
@@ -1 +1,52 @@
...
+    \item one which continues the build process on the changes code base as is; and
+    \item one which merges the changes into the master branch and continues from there;
+\end{enumberate}
+\item compiling the app and linking dependencies, such as external libraries;
+\item linting the newly added code, which entails checking the code style for bad practices and use of deprecated functionality in e.g. the Android APIs;
+\item performing tests, both unit testing but also device tests which requires running the application on an actual Android phone;
+\item generate documentation, based on the Javadoc\footnote{\url{https://en.wikipedia.org/wiki/Javadoc}} defined in comments throughout the code; and
+\item finally, packaging the Android app into a downloadable apk--file, for pre--release distribution.
+\end{enumberate}
+As described in the sub points \textit{1.1} and \textit{1.2}, the automated build system, effectively carries out two builds for every run, one of the changes in an isolated manner, and one where the changes are integrated with the current master branch.
```
Comment (TriXXeD) [inline cha/implementaion/initial/ci.tex]: use labels and ref

File: cha/implementaion/initial/ci.tex [hunk 4]
```diff
@@ -1 +1,52 @@
...
+%We want to be assisted in integrating newly developed features etc.
+%Also, automate tests (dev env / target) and docs
+%Android APK availability
+%Hand in hand with agile
+While implementing our Android application, it is important that we do not get disconnected from the end product, albeit an unfinished one.
+This means that we need some way of continuously integrating new features, bug--fixes, and other changes into the code base, whilst also reflecting them in the product.
+However, we still want to have some assurance of our product working as intended, which means that we also need to perform some form of testing.
+
+\bigskip
+To meet these needs, we utilize \textit{continuous integration}, or \textit{CI}, in our development process.
```
Comment (TriXXeD) [inline cha/implementaion/initial/ci.tex]: add CI to acronym.tex and use

**Review by TriXXeD — DISMISSED** [review]
Just clarify in regard to the first comment and it is fine.

**Pull request merged**